tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ThreadContextTest.java

```diff
@@ -1029,6 +1029,112 @@ public void withContextCaptureDependentStageForcedCompletion() throws ExecutionE
+                .build();
+        ThreadContext labelContext = ThreadContext.builder()
+                .propagated(Label.CONTEXT_NAME)
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+
+        long testThreadId = Thread.currentThread().getId();
+        try {
+            Function<Character, String> getContext = c -> Buffer.get().append(c).append(Label.get()).toString();
+
+            // Set non-default values
+            Buffer.set(new StringBuffer("withContextCaptureMultipleThreadContexts-test-buffer-A"));
+            Label.set("withContextCaptureMultipleThreadContexts-test-label-A");
+
+            CompletableFuture<Character> unmanagedStage1 = new CompletableFuture<Character>();
+
+            CompletableFuture<Character> stage2 = bufferContext.withContextCapture(unmanagedStage1);
+            CompletableFuture<Character> stage3 = labelContext.withContextCapture(unmanagedStage1);
+
+            CompletableFuture<String> stage4 = stage2.thenApply(getContext);
+            CompletableFuture<String> stage5 = stage3.thenApply(getContext);
+
+            // change context
+            Buffer.set(new StringBuffer("withContextCaptureMultipleThreadContexts-test-buffer-B"));
+            Label.set("withContextCaptureMultipleThreadContexts-test-label-B");
+
+            unmanagedStage1.complete(';');
+
+            Assert.assertEquals(stage4.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), "withContextCaptureMultipleThreadContexts-test-buffer-A;",
+                    "Context was incorrectly established on contextual function.");
+
+            Assert.assertEquals(stage5.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), ";withContextCaptureMultipleThreadContexts-test-label-A",
+                    "Context was incorrectly established on contextual function.");
+        }
+        finally {
+            // Restore original values
+            Buffer.set(null);
+            Label.set(null);
+        }
+    }
+
+    /**
+     * The withContextCapture method should be able to create a dependent stage that is
+     * associated with a thread context even if its parent stage is already associated with
+     * a different thread context. Both stages must honor their respective thread context.
+     */
+    @Test
+    public void withContextCaptureSwitchThreadContext()
+            throws ExecutionException, InterruptedException, TimeoutException {
+        ThreadContext bufferContext = ThreadContext.builder()
+                .propagated(Buffer.CONTEXT_NAME)
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+        ThreadContext labelContext = ThreadContext.builder()
+                .propagated(Label.CONTEXT_NAME)
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+
+        long testThreadId = Thread.currentThread().getId();
+        try {
+            Function<Character, String> getContext = s -> Buffer.get().append(s).append(Label.get()).toString();
```

Comment:
Did you intend to use getContext on line 1112?

Reply:
no, the getContext lambda is just another case of unused code accidentally copy/pasted. Thanks for catching this and the other two cases.  I added a commit that removes these 3 lines.